src/pepr/operator/controllers/network/generators/kubeNodes.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2024 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import { KubernetesListObject } from "@kubernetes/client-node";
+import { V1NetworkPolicyPeer, V1NodeCondition, V1NodeAddress } from "@kubernetes/client-node";
+import { K8s, kind } from "pepr";
+import { Component, setupLogger } from "../../../../logger";
+import { RemoteGenerated } from "../../../crd";
+import { anywhere } from "./anywhere";
+import { UDSConfig } from "../../../../config";
+
+const log = setupLogger(Component.OPERATOR_GENERATORS);
+
+// Maintain a set of all node internal IPs
+const nodeSet = new Set<string>();
+
+/**
+ * Initialize the node targets by fetching the current nodes in the cluster
+ * and populating the nodeSet with their Internal IPs.
+ */
+export async function initAllNodesTarget() {
+  try {
+    const nodes = await fetchKubernetesNodes();
+    nodeSet.clear();
+
+    for (const node of nodes.items) {
+      const ip = getNodeInternalIP(node);
+      if (ip) nodeSet.add(ip);
+    }
+  } catch (err) {
+    log.error("error fetching node IPs:", err);
+  }
+}
+
+/**
+ * Returns the egress CIDRs of all known nodes as network policy peers.
+ * If none are known, defaults to 0.0.0.0/0 and logs a warning.
+ */
+export function nodeCIDRs(): V1NetworkPolicyPeer[] {
+  const policies = buildNodePolicies([...nodeSet]);
+  if (policies.length > 0) return policies;
+
+  log.warn("Unable to get API server CIDR, defaulting to 0.0.0.0/0");
+  return [anywhere];
+}
+
+/**
+ * When a node is created or updated, if it's Ready, add its IP to the set,
+ * rebuild the policies, and update the NetworkPolicies.
+ */
+export async function updateKubeNodesFromCreateUpdate(node: kind.Node) {
+  const isReady = node.status?.conditions?.some(
+    (condition: V1NodeCondition) => condition.type === "Ready" && condition.status === "True",
+  );
+
+  if (isReady) {
+    const ip = getNodeInternalIP(node);
+    if (ip) nodeSet.add(ip);
+  }
```

Comment:
🤔 I didn't consider that. It might cause churn but I added a `delete()` call to the set if the node is **not** ready. We can adjust it later if needed. 

Thinking though, the target is `KubeNodes`, and not guaranteed to be _ready_ nodes I suppose? Even if right now we're trying to only include the ready ones.

Reply:
Yeah, especially in this case of metrics I suppose there could be an argument that a slightly unhealthy node is the exact type of node that you'd want to get metrics from (if they were reachable). Now I'm wondering if we should just ignore health altogether...worst case a pod has an extra egress permission to a node that is unhealthy, or partially unhealthy 🤔

Reply:
Would be curious on @UnicornChance / @noahpb opinions here...should we consider node health as part of the netpol construction? I very slightly lean towards not considering the health because pods may still want to be able to hit the node even if it's reporting unhealthy?

Reply:
I agree with @mjnagel  - a node might not appear as `Ready` when it has `DiskPressure` or `MemoryPressure`, but workloads could still be running on it. I think we should only cut off network access to the nodes once we're certain it's no longer needed.